src/sc/sidechain.cpp

```diff
@@ -783,108 +782,91 @@ void CSidechain::UpdateScFees(const CScCertificateView& certView, int blockHeigh
         while (scFees.size() > max_size)
         {
             const auto& entry = scFees.front();
-            LogPrint("sc", "%s():%d - popping f=%d, m=%d from list, as scFees maxsize has been reached\n",
-                __func__, __LINE__, entry.forwardTxScFee, entry.mbtrTxScFee);
+            LogPrint("sc", "%s():%d - popping %s from list, as scFees maxsize has been reached\n",
+                __func__, __LINE__, entry->ToString().c_str());
             scFees.pop_front();
         }
     }
     // v2 sidechains
     else {
-        auto scFeeIt = std::find_if(scFees_v2.begin(), scFees_v2.end(),
-            [&blockHeight](Sidechain::ScFeeData_v2 scFeeElem) { return scFeeElem.submissionHeight == blockHeight; });
+        auto scFeeIt = std::find_if(scFees.begin(), scFees.end(),
+            [&blockHeight](const std::shared_ptr<Sidechain::ScFeeData> & scFeeElem) {
+                return std::dynamic_pointer_cast<Sidechain::ScFeeData_v2>(scFeeElem)->submissionHeight == blockHeight;
+            });
 
         // We have not found a scFeeData for the current height, so we add a new one and perform all the checks
         // on the container size / element age
-        if (scFeeIt == scFees_v2.end()) {
-            scFees_v2.emplace_back(ftScFee, mbtrScFee, blockHeight);
+        if (scFeeIt == scFees.end()) {
+            scFees.emplace_back(new Sidechain::ScFeeData_v2(ftScFee, mbtrScFee, blockHeight));
 
             // as for v1 sidechains
             const size_t max_size { static_cast<size_t>(maxSizeOfScFeesContainers) };
-            while (scFees_v2.size() > max_size)
+
+            while (scFees.size() > max_size)
             {
-                const auto& entry = scFees_v2.front();
-                LogPrint("sc", "%s():%d - popping f=%d, m=%d, h=%d from list, as scFees maxsize has been reached\n",
-                    __func__, __LINE__, entry.forwardTxScFee, entry.mbtrTxScFee, entry.submissionHeight);
-                scFees_v2.pop_front();
+                const auto& entry = scFees.front();
+                LogPrint("sc", "%s():%d - popping %s from list, as scFees maxsize has been reached\n",
+                    __func__, __LINE__, entry->ToString().c_str());
```

Comment:
Here I think .c_str() is a bit pleonastic. Strings are already implicitly convertible to a null terminated char*

Reply:
Done